docs/fsharp/style-guide/formatting.md

```diff
@@ -8,103 +8,50 @@ ms.date: 08/31/2020
-
-Always use white space around binary arithmetic expressions:
-
-```fsharp
-let subtractThenAdd x = x - 1 + 3
-```
-
-Unary `-` operators should always be immediately followed by the value they are negating:
-
-```fsharp
-// OK
-let negate x = -x
-
-// Bad
-let negateBad x = - x
-```
-
-Adding a white-space character after the `-` operator can lead to confusion for others.
-
-In summary, it's important to always:
-
-* Surround binary operators with white space
-* Never have trailing white space after a unary operator
-
-The binary arithmetic operator guideline is especially important. Failing to surround a binary `-` operator, when combined with certain formatting choices, could lead to interpreting it as a unary `-`.
-
-### Surround a custom operator definition with white space
-
-Always use white space to surround an operator definition:
-
-```fsharp
-// OK
-let ( !> ) x f = f x
-
-// Bad
-let (!>) x f = f x
-```
-
-For any custom operator that starts with `*` and that has more than one character, you need to add a white space to the beginning of the definition to avoid a compiler ambiguity. Because of this, we recommend that you simply surround the definitions of all operators with a single white-space character.
+When indentation is required, you must use spaces, not tabs. F# code does not use tabs and the compiler will give an error if a tab character is encountered outside
+a string literal or comment.
 
-### Surround function parameter arrows with white space
+### Use consistent indentation
 
-When defining the signature of a function, use white space around the `->` symbol:
-
-```fsharp
-// OK
-type MyFun = int -> int -> string
+When indenting, at least one space is required. Your organization can create coding standards to specify the number of spaces to use for indentation; two, three, or four spaces of indentation at each level where indentation occurs is typical.
 
-// Bad
-type MyFunBad = int->int->string
-```
+**We recommend four spaces per indentation.**
 
-### Surround function arguments with white space
+That said, indentation of programs is a subjective matter. Variations are OK, but the first rule you should follow is *consistency of indentation*. Choose a generally accepted style of indentation and use it systematically throughout your codebase.
 
-When defining a function, use white space around each argument.
+### Use an automatic code formatter
 
-```fsharp
-// OK
-let myFun (a: decimal) b c = a + b + c
+The [Fantomas code formatter](https://github.com/fsprojects/fantomas/#fantomas) is the F# community standard tool for automatic code formatting. The default
+settings correspond to this style guide.
 
-// Bad
-let myFunBad (a:decimal)(b)c = a + b + c
-```
+We strongly recommend the use of this code formatter. Within F# teams, code formatting specifications should be agreed and codified in terms
+of a agreed settings for the code formatter. If the code formatter doesn't implement the setting you require, contribute a new
```

Comment:
Could you please leave out the sentence `If the code formatter doesn't implement the setting you require, contribute a new setting to the project.`. This is really not the case, our [contribution guidelines](https://github.com/fsprojects/fantomas/blob/master/CONTRIBUTING.md#what-are-we-not-looking-for) are very clear on that.

The impact of settings can be huge and we had problems in the past where people do a one-time contribution, introduce massive technical dept and are not there to clean up the mess. Each time an FCS update or editor tooling changes we need to deal with all those things. It is not worth having currently.
That is why we currently stick to two style guides and that is it.

Keep in mind that Fantomas at the time of writing cannot even fully format the F# compiler itself. That is a higher purpose to achieve than trying to keep up with everybody's little nitpicks on how stuff should look like.
It is not worth it and it burns people out. Once the compiler can be formatted, we might revisit our policy but until then we really have bigger fish to fry.

Reply:
Got it, thanks, will do